clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -3830,9 +3824,17 @@ protected Set<String> maybeSendIngestionHeartbeat() {
       // Not time for another heartbeat yet.
       return null;
     }
+    Set<String> failedPartitions = sendHeartbeatToRealtimeTopic(false);
```

Comment:
Should we call this method just once and then send HB to both reg RT and sep RT if the conditions meet.  This way we don't need to do double looping and verification. Additionally we can make return `maybeSendIngestionHeartbeat` since the return value doesn't seem to be used anywhere

Reply:
I think about that actually, then I will need to do send HB internal method manually for twice, but maybe this is more clean, I don't have a strong opinion for that.

> Additionally we can make return maybeSendIngestionHeartbeat since the return value doesn't seem to be used anywhere
I checked it is still used in test, we should keep it.

Reply:
Actually I know why I did that, it is because the below logging requires the topic to be specified...let me think if I can separate the log only and do that send in one loop